### DISCUSSION TITLE
Scan the AI models

### DISCUSSION THREAD

**Discussion by wy65701436** [aquasecurity/trivy#7148] [Ideas]
### Description

Currently, I am working on packaging an AI model, which I downloaded from Hugging Face, into the OCI format and pushing it to Harbor. As a next step, I want to enable the default scanner, Trivy, to scan my AI model. However, it seems that Trivy does not support scanning AI models in this scenario.

I am wondering if Trivy has any plans to support scanning AI models built with the OCI format and report on license and CVE issues for these models.

### Target

None

### Scanner

None

**Comment by knqyf263** [comment]
As far as I know, the AI model is a serialized binary. Is there any way to detect used libraries?

**Reply by chen-keinan** [reply]
ref: https://jfrog.com/blog/data-scientists-targeted-by-malicious-hugging-face-ml-models-with-silent-backdoor/

**Reply by knqyf263** [reply]
Yes, it's more like SAST. I'm wondering how SCA scans AI models for vulnerabilities.

**Reply by nikpivkin** [reply]
Malicious bytecode can be injected into the pickle file where weights are stored. But now the [safetensors](https://github.com/huggingface/safetensors)  format for storing model weights is widely used, which does not have this problem. https://huggingface.co/blog/safetensors-security-audit
